### PR TITLE
Added in contents api

### PR DESCRIPTION
**Pull request by jlewis91**
Built out the contents api section of the github api, where files can be downloaded / added / updated / deleted. Added in sketch of tests for this section, though it's unclear how best to implement these (given the credentials), so left commented out...

### CODE REVIEW

**Comment by WestleyArgentum** [comment]
Hey! Sorry for the delay, been swamped the last few days.

File: src/contents.jl [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            get(data, "parents", nothing),
+            get(data, "message", nothing),
+            get(data, "tree", nothing),
+            get(data, "type", nothing))
+    end
+
+end
+
+
+# function contents(repo::Repo; auth = AnonymousAuth(), options...)
```
Comment (WestleyArgentum) [inline src/contents.jl]: Why are these commented out - do they work?

File: src/contents.jl [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    authenticate_headers(headers, auth)
+    r = get(URI(API_ENDPOINT; path = "/repos/$owner/$repo/readme");
+                    headers = headers)
+    handle_error(r)
+    readme_file = File(JSON.parse(r.data))
+    return readme_file
+end
+
+## Issues
+#4) need to add get data method to files with links...
```
Comment (WestleyArgentum) [inline src/contents.jl]: Is this something your still working on adding to this PR? If not maybe we should remove it here and file it as an issue.

File: src/contents.jl [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+#5) need to fix some fields in files & commit types
```
Comment (WestleyArgentum) [inline src/contents.jl]: Same here -- does this work right now or should we fix it before merge?

File: src/contents.jl [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+#6) Implement Owner, etc classes in new functions
```
Comment (WestleyArgentum) [inline src/contents.jl]: There are lots of functions that don't quite handle this well right now, I'm guilty of writing most of them. There's an umbrella issue: https://github.com/WestleyArgentum/GitHub.jl/issues/25 so we can probably remove it from the code here.

File: src/contents.jl [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    data = {"message" => message, "sha" => sha}
+    branch == "default" || (data["branch"] = branch)
+    author.name == "NA" || (data["author"] = author)
+    committer.name == "NA" || (data["committer"] = committer)
+    authenticate_headers(headers, auth)
+    r = Requests.delete(URI(API_ENDPOINT;
+            path = "/repos/$owner/$repo/contents/$path"),
+            json=data, headers = headers)
+    handle_error(r)
+    # print(get(JSON.parse(r.data), "commit", nothing))
```
Comment (WestleyArgentum) [inline src/contents.jl]: :) not a big deal, obviously, but you might want to remove this debugging line as well

**Comment by WestleyArgentum** [comment]
Looks really awesome overall, it'll be really nice being able to use julia for this!

File: src/contents.jl (hunk 5 above)
Comment (jlewis91) [inline src/contents.jl]: Thanks. Fixed.

File: src/contents.jl (hunk 4 above)
Comment (jlewis91) [inline src/contents.jl]: Sounds good.

**Comment by jlewis91** [comment]
Build will fail because tests require authorization token...the tests passed on my local system; is there a good solution for this?

File: src/contents.jl (hunk 1 above)
Comment (jlewis91) [inline src/contents.jl]: these work, but no tests written for them.

File: src/contents.jl (hunk 2 above)
Comment (jlewis91) [inline src/contents.jl]: perfect, see #31

**Comment by WestleyArgentum** [comment]
Great, thanks for updating all of that. I think this is good to merge into master -- that way it will be easier for people to start playing with.

As for tests, I didn't know about travis supporting encryption, my first thought was to just create a test account like you have set up now. For tonight I might just comment things out (otherwise the julia package robot will get angry), but I'll come back and implement a better solution as soon as I get the chance. Thanks for finding those docs!

**Pull request merged**